test/integration/integration_test.py

```diff
@@ -0,0 +1,117 @@
+}
+
+image_reg = {compile(r"localhost:12345/sampleapp\s+local.+"): False}
+
+pods_regs = {
+    compile(r".*mqtt-broker-.+-.+\s+1/1\s+Running.*"): False,
+    compile(r".*bash-.+-.+\s+1/1\s+Running.*"): False,
+    compile(r".*seatservice-.+-.+\s+2/2\s+Running.*"): False,
+    compile(r".*feedercan-.+-.+\s+2/2\s+Running.*"): False,
+    compile(r".*vehicledatabroker-.+-.+\s+2/2\s+Running.*"): False,
+    compile(r".*zipkin-.+-.+\s+1/1\s+Running.*"): False,
+    compile(r".*sampleapp-.+-.+\s+2/2\s+Running.*"): False,
+}
+
+uninstall_runtime_reg = {
+    compile(r"release \"vehicleappruntime\" uninstalled"): False
+}
+
+reset_controlplane_reg = {
+    compile(r"Successfully deleted cluster cluster!"): False
+}
+
+
+def escape_ansi(line):
+    ansi_escape = compile(r'(\x9B|\x1B\[)[0-?]*[ -\/]*[@-~]')
+    return ansi_escape.sub('', line)
+
+
+def check_regs(regs: dict, output: str) -> bool:
+    final_flag = True
+    # optional print
+    print(output)
+    for reg in regs:
+        if not regs[reg]:
+            if reg.search(output):
+                regs[reg] = True
+        final_flag = final_flag and regs[reg]
+    return final_flag
+
+
+def check_cluster_status_and_dapr(kubectl_regs: dict, dapr_regs: dict) -> bool:
+    command_logs_kubectl = check_output(["kubectl", "cluster-info"]).decode()
+    command_logs_dapr = check_output(["dapr", "status", "-k"]).decode() 
+    command_logs_kubectl = escape_ansi(command_logs_kubectl)
+    return check_regs(kubectl_regs, command_logs_kubectl) and check_regs(dapr_regs, command_logs_dapr)
+
+
+def deploy_runtime_check(kubectl_regs: dict, dapr_regs: dict) -> bool:
+    command_logs_kubectl = check_output(["kubectl", "cluster-info"]).decode()
+    command_logs_dapr = check_output(["dapr", "status", "-k"]).decode() 
+    command_logs_kubectl = escape_ansi(command_logs_kubectl)
+    return check_regs(kubectl_regs, command_logs_kubectl) and check_regs(dapr_regs, command_logs_dapr)
+
+
+def check_image_if_created(image_reg: Pattern) -> bool:
+    command_get_images_logs = check_output(["docker", "images"]).decode()
+    return check_regs(image_reg, command_get_images_logs)
+
+
+def check_pods(pods_regs: dict) -> bool:
+    command_get_pods_logs = check_output(["kubectl", "get", "pods"]).decode()
+    return check_regs(pods_regs, command_get_pods_logs)
+
+
+def reset_controlplane(reset_control_reg: dict) -> bool:
+    command_reset_controlplane_logs = check_output(base_command_runtime.split(" ") + ["reset_controlplane"]).decode()
+    return check_regs(reset_controlplane_reg, command_reset_controlplane_logs)
+
+
+def uninstall_runtime(uninstall_runtime_reg: dict) -> bool:
+    command_uninstall_runtime_logs = check_output(base_command_runtime.split(" ") + ["uninstall_runtime"]).decode()
+    return check_regs(uninstall_runtime_reg, command_uninstall_runtime_logs)
+
+
+def run_command(command: str) -> bool:
+    return (0 == check_call(command, shell=True))
```

Comment:
Is there any specific reason why `shell` is set to `true`? If it is not needed, we should try to avoid it [see](https://medium.com/@acharya.vikash/avoid-using-shell-true-in-python-subprocess-module-e95fed487f19)